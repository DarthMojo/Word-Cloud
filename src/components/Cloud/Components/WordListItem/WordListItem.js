import React from "react";
import type { Element, Children } from "react";

type WordListItemType = {
    id: string,
    x: number,
    y: number,
    fontSize: number,
    fontFamily: string,
    children: Children,
    color: string,
    onWordItemClick: Function,
    rotate: number,
    textAnchor: string
};

/**
 *
 * @param id
 * @param x
 * @param y
 * @param fontSize
 * @param color
 * @param fontFamily
 * @param children
 * @param rotate
 * @param textAnchor
 * @param onWordItemClick
 * @return {Element}
 * @constructor
 */
const WordListItem =
    ({
        id, x, y, fontSize, color, fontFamily, children,
        rotate = 0, textAnchor = "middle", onWordItemClick
    }: WordListItemType): Element<any> => {
        return (
            <text
                className="c-text"
                style={{ fontSize, fontFamily, fill: color, cursor: "pointer" }}
                transform={`translate(${x}, ${y}) rotate(${rotate})`}
                textAnchor={textAnchor}
                onClick={(): void => onWordItemClick(id)}>
                {children}
            </text>
        );
    };

export default WordListItem;
