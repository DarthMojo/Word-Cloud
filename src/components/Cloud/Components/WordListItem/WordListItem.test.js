import React from "react";
import data from "../../../../config/data/topics.json";
import WordListItem from "./WordListItem";
import { shallowToJson } from "enzyme-to-json";

test("render WordListItem()", async () => {
    const onWordItemClick = () => {};

    const wordItem = await data.topics.find(wordItem => wordItem.id === "1751295897__Berlin");

    const wrapper = shallow(
        <WordListItem
            id={wordItem.id}
            x={98}
            y={175}
            fontSize={17}
            color={"#49E20E"}
            fontFamily={"Impact, sans-serif"}
            rotate={90}
            textAnchor={"middle"}
            onWordItemClick={onWordItemClick}
            children={wordItem.label} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});
