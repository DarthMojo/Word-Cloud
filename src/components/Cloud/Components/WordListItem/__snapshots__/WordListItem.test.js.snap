// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render WordListItem() 1`] = `
<text
  className="c-text"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "fill": "#49E20E",
      "fontFamily": "Impact, sans-serif",
      "fontSize": 17,
    }
  }
  textAnchor="middle"
  transform="translate(98, 175) rotate(90)"
>
  Berlin
</text>
`;
